File: app/src/main/java/com/example/administrator/myapplication/MainModule.java
package com.example.administrator.myapplication;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Administrator on 2017/11/14.
 * MainModule
 */
@Module
class MainModule {

    private final MainContract.View mView;

    MainModule(MainContract.View view) {
        mView = view;
    }

    @Provides
    MainContract.View provideMainView() {
        return mView;
    }

    @Provides
    MainPresenter provideMainPresenter(User user){
        return new MainPresenter(mView,user);
    }

    @Provides
    User provideUser() {
        return new User("你好我是注入");
    }
}

File: app/src/main/java/com/example/administrator/myapplication/MainModule.kt
package com.example.administrator.myapplication

import dagger.Module
import dagger.Provides

/**
 * Created by Administrator on 2017/11/14.
 * MainModule
 */
@Module
internal class MainModule(private val mView: MainContract.View) {

    @Provides
    fun provideMainView(): MainContract.View = mView

    @Provides
    fun provideMainPresenter(user: User): MainPresenter = MainPresenter(mView, user)

    @Provides
    fun provideUser(): User = User("你好我是注入")
}
